pkg/cmd/server/api/types.go

```diff
@@ -699,3 +699,93 @@ type AssetExtensionsConfig struct {
 	// Web Console's context root. Defaults to false.
 	HTML5Mode bool
 }
+
+type LDAPSyncConfig struct {
+	api.TypeMeta
+	// LDAPClientInfo contains the information about connection to the LDAP server.
+	// The contained URL will be parsed for scheme, host and port. The syntax of the URL is:
+	// `scheme://host:port/`, where `scheme` is either `ldap` or `ldaps`. Any other encoding in the
+	// URL will be ignored.
+	LDAPClientInfo LDAPPasswordIdentityProvider
+
+	// GroupNameAttributes (optionally) maps LDAP group entries to OpenShift Groups
+	GroupNameAttributes []string
```

Comment:
@deads2k @liggitt I need your input on this - as it stands I chose not to include this optional and schema-specific field in the QueryTemplates because I really liked how that was a non-negotiable part of the sync config - regardless of what schema you have, the query templates will provide the query templates. I didn't quite like to have a catch-all "schema-specific-config" object to shove something like this in. Should it remain as an optional field here or should there be another `EmbeddedObject` to hold other schema-specific config? Should there be just one catch-all schema-specific config object?